src/server/game/Handlers/ItemHandler.cpp

```diff
@@ -440,6 +440,13 @@ void WorldSession::HandleSellItemOpcode(WorldPacket& recvData)
         ItemTemplate const* pProto = pItem->GetTemplate();
         if (pProto)
         {
+            uint32 money = pProto->SellPrice * count;
+            if (_player->GetMoney() + money > MAX_MONEY_AMOUNT)               // prevent exceeding gold limit
```

Comment:
Check similar code like https://github.com/TrinityCore/TrinityCore/blob/0d8672732552b0474e22b077a2318fa87446f25f/src/server/game/Handlers/TradeHandler.cpp#L299 to make sure `_player->GetMoney() + money` doesn't overflow (and please test it with the overflow case to verify it works as expected)

Reply:
yep, it has trouble when item costs 1 copper

Reply:
fixed